worlds/LauncherComponents.py

```diff
@@ -18,16 +18,39 @@ class Type(Enum):
 
 
 class Component:
+    """
+    A Component represents a process launchable by Archipelago Launcher, either by a User action in the GUI,
+    by resolving an archipelago://user:pass@host:port link from the Webhost, by resolving a patch file's metadata
+    or by using a component name arg while running the Launcher in CLI i.e. `ArchipelagoLauncher.exe "Text Client"`
+
+    Expected to be appended to LauncherComponents.component list to be used.
+    """
     display_name: str
+    """Used as the GUI button label and the component name in the CLI args"""
     type: Type
+    """Classification of component intent to filter in the Launcher GUI"""
```

Comment:
A little confused by what this is actually supposed to mean. Looking at nearby code helps a little, but after reading this, I'm not fully sure what type DOES, if I should set it, what I should set it to/with, and what happens if I choose one option over the others.

Reply:
It affects the column and sort order of the Launcher UI.

Right side is Clients, left side is Tools, Misc, and Adjusters in that order.

Reply:
I guess my point is that it's not obvious when reading the docstring. At the last, I would expect that making a component a client puts it on the right side.

Reply:
part of my intent on this wording is to be generic enough that it doesn't need to be rewritten with #3934
is the typing of type being Type the confusing part? If it was more obvious that there was an enum to reference about potential values help clarify?

Reply:
Also when writing it I forgot that the init parameter for type is optional (and gets inferred by substrings of component name), because the actual value on the class is not optional
so that clarification could be expanded upon

Reply:
Referencing the enum could help

Reply:
updated with a bit more clarifications